js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+
+- An Amazon DynamoDB table with 5 r/w units. Support for auto-scaling and encryption at rest coming soon.
+- An AWS AppSync DataSource configured to access the table above.
+- An AWS IAM role attached to the DataSource that allows AWS AppSync to call the above table on your behalf.
+- Up to 8 resolvers (create, update, delete, get, list, onCreate, onUpdate, onDelete) but this is configurable via the `query`, `mutation`, and `subscription` arguments on the `@model` directive.
+
+This input schema document
+
+```
+type Post @model {
+    id: ID!
+    title: String
+}
+type MetaData {
+    category: Category
+}
+enum Category { comedy news }
+```
+
+would generate the following schema parts
+
+```
+type Post {
+  id: ID!
+  title: String!
+  metadata: MetaData
+}
+
+type MetaData {
+  category: Category
+}
+
+enum Category {
+  comedy
+  news
+}
+
+input MetaDataInput {
+  category: Category
+}
+
+enum ModelSortDirection {
+  ASC
+  DESC
+}
+
+type ModelPostConnection {
+  items: [Post]
+  nextToken: String
+}
+
+input ModelStringFilterInput {
+  ne: String
+  eq: String
+  le: String
+  lt: String
+  ge: String
+  gt: String
+  contains: String
+  notContains: String
+  between: [String]
+  beginsWith: String
+}
+
+input ModelIDFilterInput {
+  ne: ID
+  eq: ID
+  le: ID
+  lt: ID
+  ge: ID
+  gt: ID
+  contains: ID
+  notContains: ID
+  between: [ID]
+  beginsWith: ID
+}
+
+input ModelIntFilterInput {
+  ne: Int
+  eq: Int
+  le: Int
+  lt: Int
+  ge: Int
+  gt: Int
+  contains: Int
+  notContains: Int
+  between: [Int]
+}
+
+input ModelFloatFilterInput {
+  ne: Float
+  eq: Float
+  le: Float
+  lt: Float
+  ge: Float
+  gt: Float
+  contains: Float
+  notContains: Float
+  between: [Float]
+}
+
+input ModelBooleanFilterInput {
+  ne: Boolean
+  eq: Boolean
+}
+
+input ModelPostFilterInput {
+  id: ModelIDFilterInput
+  title: ModelStringFilterInput
+  and: [ModelPostFilterInput]
+  or: [ModelPostFilterInput]
+  not: ModelPostFilterInput
+}
+
+type Query {
+  getPost(id: ID!): Post
+  listPosts(filter: ModelPostFilterInput, limit: Int, nextToken: String): ModelPostConnection
+}
+
+input CreatePostInput {
+  title: String!
+  metadata: MetaDataInput
+}
+
+input UpdatePostInput {
+  id: ID!
+  title: String
+  metadata: MetaDataInput
+}
+
+input DeletePostInput {
+  id: ID
+}
+
+type Mutation {
+  createPost(input: CreatePostInput!): Post
+  updatePost(input: UpdatePostInput!): Post
+  deletePost(input: DeletePostInput!): Post
+}
+
+type Subscription {
+  onCreatePost: Post @aws_subscribe(mutations: ["createPost"])
+  onUpdatePost: Post @aws_subscribe(mutations: ["updatePost"])
+  onDeletePost: Post @aws_subscribe(mutations: ["deletePost"])
+}
+```
+
+### @auth
+
+Object types that are annotated with `@auth` are protected by one of the
+supported authorization strategies. Types that are annotated with `@auth`
+must also be annotated with `@model`. Currently, Amazon Cognito user pools
+is the only supported authorization mode.
```

Comment:
API Key also, but you could probably remove this sentence so that you don't have to worry about updating it later.